src/components/chip/selectable-chip/index.tsx

```diff
@@ -0,0 +1,95 @@
+import {
+  StandaloneChevronDownRegularIcon,
+  StandaloneCircleXmarkRegularIcon,
+} from '@deriv/quill-icons'
+import { forwardRef } from 'react'
+import {
+  ChipIconSizes,
+  chipBaseClassnames,
+  ChipStandaloneIconSizes,
+  chipBaseVariant,
+  chipSelectionClassnames,
+} from '../chip.classnames'
+import { SelectableChipProps } from '../types'
+import qtMerge from 'qtMerge'
+
+export const SelectableChip = forwardRef<
+  HTMLButtonElement,
+  SelectableChipProps
+>(
+  (
+    {
+      className,
+      children,
+      dismissible = false,
+      icon: Icon,
+      isDropdownOpen,
+      dropdown = false,
+      labelTag,
+      onChipSelect,
+      onDismiss,
+      size,
+      ...rest
+    },
+    ref,
+  ) => {
+    const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+      if (dismissible || dropdown) return
+      const target = event.currentTarget
+      const isSelected = target.getAttribute('data-state') === 'selected'
+      const selected_state = isSelected ? '' : 'selected'
+      target.setAttribute('data-state', selected_state)
+      onChipSelect?.(event, !isSelected)
+    }
+
+    const handleDismiss = (
+      event: React.MouseEvent<SVGSVGElement, MouseEvent>,
+    ) => {
+      onDismiss?.(event)
+    }
+
+    return (
+      <button
+        className={chipBaseVariant({
+          dismissible,
+          size,
+          className: qtMerge(
+            chipBaseClassnames,
+            chipSelectionClassnames,
+            className,
+          ),
+        })}
+        data-state=""
+        ref={ref}
+        onClick={handleClick}
+        {...rest}
+      >
+        {Icon && <Icon {...ChipIconSizes[size ?? 'md']} />}
+        {children}
+        {labelTag && <span className="font-bold">{labelTag}</span>}
+        {dismissible && (
+          <StandaloneCircleXmarkRegularIcon
+            {...ChipStandaloneIconSizes[size ?? 'md']}
+            onClick={handleDismiss}
+            data-testid="dt-chip-dismissable-btn"
+            className="cursor-pointer"
+          />
+        )}
+        {dropdown && (
+          <>
+            <StandaloneChevronDownRegularIcon
+              data-state={isDropdownOpen ? 'open' : 'close'}
+              id="selectable_chip_chevron"
+              className="transition-transform duration-300 data-[state=open]:rotate-180"
+              {...ChipStandaloneIconSizes[size ?? 'md']}
```

Comment:
@shafin-deriv, is it possible to give `size` a default value as opposed to using the `??`?

Reply:
@jim-deriv we are currently using CVA and the size type is coming from there which can be null. Even if I put the default value it still throws an error on this line so we have to keep it this way (in other components as well).

Reply:
@shafin-deriv, have you tried `Exlude< VariantProps<typeof chipBaseVariant >, null> ? This should work I think

Reply:
@jim-deriv yes, I agree. We tried to get around this, but couldn't find anything yet. `Exclude<VariantProps<typeof chipBaseVariant>, null>` may not work directly in this case but I got your point. Let me dig into this a bit more. I will share in the channel if I find anything. Thanks for the suggestion 👍

Reply:
@jim-deriv done now ✅ Thanks for the idea!!!